### PR TITLE
Adds the ability to run SwiftFormat from Swift PM locally

### PR DESCRIPTION
**Pull request by orta**
Heyo 👋 

This allows you to install SwiftFormat via Swift PM and have the extension automatically pick it up. It does this by checking the default paths for where the built files would be.

Here it is in action:

format 2018-11-24 09_02_02

I gave this entire project a bit of a spit shine while I was looking at adding a feature. These were good solid foundations to work on 👍

### CODE REVIEW

**Comment by johndpope** [comment]
Hi @orta,
did you see this vscode editor by apple with source kit lsp?
 https://github.com/apple/sourcekit-lsp/tree/master/Editors

**Comment by vknabel** [comment]
Hey @orta 
Thank you for your contribution and especially for all those little changes! It looks really good. I will have look on this tomorrow.

**Comment by orta** [comment]
@johndpope  - yeah I *only* used that to [build Komondor](https://github.com/orta/Komondor) - it's looking real good. 

@vknabel  - I know what it's like to be sherlocked, I think we're all gonna end up with an awesome vscode dev experience in a few months now that apple are throwing serious money/time at the LSP

Also, no worries, you got this

**Comment by orta** [comment]
This PR is blocked until #324 is merged - as the capitalization needs to change probably

**Comment by vknabel** [comment]
@orta that's why I really love Apple's decision and I hope being sherlocked felt that great for you, too. Apple has much more resources and knows SourceKit much better than I do. So I gain more time and better tools. 😍

File: src/Current.ts [hunk 1]
```diff
@@ -66,23 +67,36 @@ export function prodEnvironment(): Current {
     config: {
       isEnabled: () =>
         vscode.workspace.getConfiguration().get("swiftformat.enable"),
+      swiftFormatPath: (document: vscode.TextDocument) => {
+        // Support running from Swift PM projects
+        const possibleLocalPaths = [".build/debug/swiftformat", ".build/release/swiftformat"]
```
Comment (segiddins) [inline src/Current.ts]: Should release be preferred to debug, since it’ll probably be faster?

**Pull request merged**